ui/app/components/http-requests-bar-chart.js

```diff
@@ -0,0 +1,206 @@
+import Component from '@ember/component';
+import d3 from 'd3-selection';
+import d3Scale from 'd3-scale';
+import d3Axis from 'd3-axis';
+import d3TimeFormat from 'd3-time-format';
+import { assign } from '@ember/polyfills';
+import { computed } from '@ember/object';
+import { run } from '@ember/runloop';
+import { task, waitForEvent } from 'ember-concurrency';
+
+/**
+ * @module HttpRequestsBarChart
+ * HttpRequestsBarChart components are used to render a bar chart with the total number of HTTP Requests to a Vault server per month.
+ *
+ * @example
+ * ```js
+ * <HttpRequestsBarChart @counters={{counters}} />
+ * ```
+ *
+ * @param counters=null {Array} - A list of objects containing the total number of HTTP Requests for each month. `counters` should be the response from the `/internal/counters/requests` endpoint which looks like:
+ * COUNTERS = [
+ *    {
+ *       "start_time": "2019-05-01T00:00:00Z",
+ *       "total": 50
+ *     }
+ * ]
+ */
+
+const COUNTERS = [
+  {
+    start_time: '2019-05-01T00:00:00Z',
+    total: 50000,
+  },
+  {
+    start_time: '2019-04-01T00:00:00Z',
+    total: 4500,
+  },
+  {
+    start_time: '2019-03-01T00:00:00Z',
+    total: 550000,
+  },
+];
+
+export default Component.extend({
+  classNames: ['http-requests-bar-chart-container'],
+  counters: COUNTERS,
+  svgContainer: null,
+  barsContainer: null,
+  clipPathRect: null,
+  margin: { top: 24, right: 16, bottom: 24, left: 16 },
+  width: 0,
+  height() {
+    const { margin } = this;
+    return 240 - margin.top - margin.bottom;
+  },
+
+  parsedCounters: computed('counters', function() {
+    // parse the start times so bars and ticks display properly
+    const { counters } = this;
+    return counters.map(counter => {
+      return assign({}, counter, { start_time: d3TimeFormat.isoParse(counter.start_time) });
+    });
+  }),
+
+  yScale: computed('parsedCounters', 'height', function() {
+    const { parsedCounters } = this;
+    const height = this.height();
+    const counterTotals = parsedCounters.map(c => c.total);
+
+    return d3Scale
+      .scaleLinear()
+      .domain([0, Math.max(...counterTotals)])
+      .range([height, 0]);
+  }),
+
+  xScale: computed('parsedCounters', 'width', function() {
+    const { parsedCounters, width, margin } = this;
+
+    return d3Scale
+      .scaleBand()
+      .domain(parsedCounters.map(c => c.start_time))
+      .rangeRound([0, width - margin.left - margin.right], 0.05)
+      .paddingInner(0.04);
```

Comment:
Would it make sense to pull padding and margin values into a global const or an object on the component? This feels a big magic number-y.

Reply:
you're right -- the padding here is definitely magic number-y. i thought about storing it as a property on the component, but decided against it because it wasn't used anywhere else and the value `0.04` is relative to the size of the d3 band (it reflects 4% of the size of the band). i was hesitant about storing that value outside of this scope because it really shouldn't be used  elsewhere.

my other thought was that `margin: { top: 24, left: 16 }` looks more like css -- that is, it's easy to see that this would correspond to 24px, 16px etc. however (and maybe this is just me overthinking/noticing that we use pixels more often than percentages in our css) `padding: 0.04` takes me a moment longer to figure out that it's a percentage. 

anyway, if i'm missing something or you think setting the padding as a property would improve legibility i'm totally happy to change it. just let me know!

Reply:
jk -- i ended up needing to use the padding again, so i moved the value into the component. thank you for the feedback, it was a helpful concept to think through!